packages/instrumentation-neo4j/src/neo4j.ts

```diff
@@ -0,0 +1,135 @@
+import { SpanStatusCode, diag, getSpan, context } from '@opentelemetry/api';
+import { DatabaseAttribute } from '@opentelemetry/semantic-conventions';
+import { VERSION } from './version';
+import type * as neo4j from 'neo4j-driver';
+import {
+    InstrumentationBase,
+    InstrumentationModuleDefinition,
+    InstrumentationNodeModuleFile,
+    InstrumentationNodeModuleDefinition,
+    safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+import { Neo4jInstrumentationConfig } from './types';
+import { getAttributesFromNeo4jSession } from './utils';
+
+type Neo4J = typeof neo4j;
+
+export class Neo4jInstrumentation extends InstrumentationBase<Neo4J> {
+    static readonly component = 'neo4j-driver';
+    protected _config!: Neo4jInstrumentationConfig;
+
+    constructor(config: Neo4jInstrumentationConfig = {}) {
+        super('opentelemetry-instrumentation-neo4j', VERSION, Object.assign({}, config));
+    }
+
+    setConfig(config: Neo4jInstrumentationConfig = {}) {
+        this._config = config;
+    }
+
+    protected init(): InstrumentationModuleDefinition<Neo4J> {
+        const apiModuleFiles = ['session', 'transaction'].map(
+            (file) =>
+                new InstrumentationNodeModuleFile<neo4j.Session>(
+                    `neo4j-driver/lib/${file}.js`,
+                    ['*'],
+                    this.patchSessionOrTransaction.bind(this),
+                    this.unpatchSessionOrTransaction.bind(this)
+                )
+        );
+
+        const module = new InstrumentationNodeModuleDefinition<Neo4J>(
+            Neo4jInstrumentation.component,
+            ['*'],
+            undefined,
+            undefined,
+            apiModuleFiles
+        );
+
+        return module;
+    }
+
+    private patchSessionOrTransaction(
+        fileExport: { default: () => neo4j.Session | neo4j.Transaction },
+        moduleVersion: string
+    ) {
+        const self = this;
+        this._wrap(fileExport.default.prototype, 'run', (originalRun: neo4j.Session['run']) => {
+            return function (query: string) {
+                if (self._config?.ignoreOrphanedSpans && !getSpan(context.active())) {
+                    return originalRun.apply(this, arguments);
+                }
+
+                const connectionAttributes = getAttributesFromNeo4jSession(this);
+                const operation = query.split(/\s+/)[0];
```

Comment:
this query will return empty operation: `'                            MATCH (p:Person) RETURN p'`.
need to clean white spaces from the query start